lib/Component.php

```diff
@@ -346,6 +348,55 @@ function jsonSerialize() {
     }
 
     /**
+     * This method serializes the data into XML. This is used to create xCard or
+     * xCal documents.
+     *
+     * @param Xml\Writer $writer  XML writer.
+     * @return void
+     */
+    function xmlSerialize(Xml\Writer $writer) {
+
+        $components = [];
+        $properties = [];
+
+        foreach($this->children as $child) {
+            if ($child instanceof Component) {
+                $components[] = $child;
+            } else {
+                $properties[] = $child;
+            }
+        }
+
+        $writer->startElement(strtolower($this->name));
+
+        if (!empty($properties)) {
+
+            $writer->startElement('properties');
```

Comment:
So, I cannot use `Xml::writeElement` here because we loop over an array, and `Xml::write`  supports only array of (`name`, `value`, `attributes`) tuples: It does not support an array of `XmlSerializable` objects. But I have update all the other parts of the code. See https://github.com/fruux/sabre-vobject/commit/6b6361088384b79b42fef16306e14abd2b4fb70f.